Cython/Compiler/ModuleNode.py

```diff
@@ -136,10 +138,15 @@ def merge_in(self, tree, scope, merge_scope=False):
             # merged in nodes should keep their original compiler directives
             # (for example inline cdef functions)
             tree = Nodes.CompilerDirectivesNode(tree.pos, body=tree, directives=scope.directives)
+        # Utility code is inserted before user code (but is inserted in the order it is defined)
+        # This makes sense because the user code depends on the utility code, so potentially relies
+        # on it being fully executed
```

Comment:
This change is worth noting (because it's potentially a fairly big change). I think it makes sense for the reasons explained in the comment.

For most utility code it doesn't really matter because it's just static declarations, but for anything that needs to be initialized in the module-init function it could create subtle bugs because the utility code hadn't executed at the point that the user code was run.